### DISCUSSION TITLE
Should we rename this repository/package?

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#645] [Ideas]
So there's a couple of reasons to consider a rename.

1) **SEO**
   - Right now it's a bit hard to find IDOM. For me personally, it was really difficult initially finding this package.
2) **Name Clarity**
   - Some beginners won't immediately know what `IDOM` does without first knowing what a _Documented Oriented Model_ is, so they might not click on the project. But, IDOM's usage is simple enough to where the user doesn't actually need to understand what a _DOM_ is.

Here's a couple potential names that could increase name clarity.
- ReactPy (My personal favorite)
- ReactiPy
- Python-React
- Reactive

I personally believe having React in the name makes sense for us, since we heavily rely on the ReactJS Client to do anything in the front-end.

Comment below if you have other naming ideas, have a personal favorite name, or if you agree/disagree with this.

If we perform a name change, this should be done sooner rather than later. The longer a package exists, the more difficult a name change becomes.

**Comment by yuvipanda** [comment]
+1 to a name change! I'm not super involved in the project, but I do agree that a different name that implies the *reactive* nature of this project will help.

**Reply by rmorshea** [reply]
Thanks for the suggestion Yuvi!

**Comment by rmorshea** [comment]


**Reply by rmorshea** [reply]
https://github.com/pypa/pypi-support/issues/1658

**Reply by rmorshea** [reply]
Any thoughts on domains if the PEP-541 request goes through? reactivepython.dev is available and somewhat reasonable.

**Reply by Archmonger** [reply]
- reactive .wiki/run/page
- reactivepython .net/dev/io
- reactive-python .com/net/io/dev/tech

I'd do the last one since a TLD is super cheap with a hyphenated URL

**Reply by rmorshea** [reply]
reactive-python.com seems good to me

**Reply by rmorshea** [reply]
Just grabbed https://github.com/reactive-python

**Reply by rmorshea** [reply]
Got a response back on the PEP-541, looks like this user has a couple unused packages and is unreachable. Might gain control of the name in the next week or two.

**Reply by rmorshea** [reply]
Ownership transfer for [reactive](https://pypi.org/project/reactive/) went through :tada:

**Reply by Archmonger** [reply]
Cheers to that 🍻

**Comment by rmorshea** [comment]
We'll also need new branding. A first pass logo could be the react atom except the loops around the atomic core are one or more snakes.

**Comment by acivitillo** [comment]
Do you plan to keep the dependency on ReactJS? I am asking because it would be very convenient if we could load other JS frameworks like Alpine.js.

**Reply by rmorshea** [reply]
IDOM's custom component interface doesn't require you to use React. So long as you appropriately implement a [`bind()`](https://idom-docs.herokuapp.com/docs/escape-hatches/javascript-components.html#custom-javascript-components) function, you can do whatever you want. The template repo uses React (which we should have a VanillaJS one too) but you could remove that if you wanted.
